src/Illuminate/Container/Container.php

```diff
@@ -791,7 +792,11 @@ public function build($concrete)
             return $concrete($this, $this->getLastParameterOverride());
         }
 
-        $reflector = new ReflectionClass($concrete);
+        try {
+            $reflector = new ReflectionClass($concrete);
+        } catch (ReflectionException $e) {
+            throw new BindingResolutionException("Target class [$concrete] does not exist.");
```

Comment:
Can we pass `ReflectionException $e` as `BindingResolutionException` `$previous`?

Reply:
Thank for this proposition, it is a good idea! This is done with 9ac88d7a938deeae09584ccb35f69dc6c37af153.